products/opensuse/main.pm

```diff
@@ -145,92 +146,107 @@ if (get_var("WITH_UPDATE_REPO")
+    }
+    die "YAML expression failed to evaluate: $@" if $@;
 }
 
-sub load_fixup_firewall {
-    # The openSUSE 13.1 GNOME disk image has the firewall disabled
-    # Upon upgrading to a new system the service state is supposed to remain as pre-configured
-    # If the service is disabled here, we enable it here
-    # For the older openSUSE base images we also see a problem with the
-    # firewall being disabled since
-    # https://build.opensuse.org/request/show/483163
-    # which seems to be in openSUSE Tumbleweed since 20170413
-    return unless get_var('HDDVERSION', '') =~ /openSUSE-(13.1-gnome)/;
-    loadtest 'fixup/enable_firewall';
-}
+=head2 load_module
 
-sub load_consoletests_minimal {
-    return unless (is_staging() && get_var('UEFI') || is_gnome_next || is_krypton_argon);
-    # Stagings should test yast2-bootloader in miniuefi at least but not all
-    loadtest "console/system_prepare";
-    loadtest "console/consoletest_setup";
-    loadtest "console/textinfo";
-    loadtest "console/hostname";
-    if (!get_var("LIVETEST")) {
-        loadtest "console/yast2_bootloader";
-    }
-    loadtest "console/consoletest_finish";
-}
-
-sub load_otherDE_tests {
-    if (get_var("DE_PATTERN")) {
-        my $de = get_var("DE_PATTERN");
-        loadtest "console/system_prepare";
-        loadtest "console/consoletest_setup";
-        loadtest "console/hostname";
-        loadtest "update/zypper_clear_repos";
-        loadtest "console/install_otherDE_pattern";
-        loadtest "console/consoletest_finish";
-        loadtest "x11/${de}_reconfigure_openqa";
-        loadtest "x11/reboot_icewm";
-        # here comes the actual desktop specific test
-        if ($de =~ /^awesome$/)       { load_awesome_tests(); }
-        if ($de =~ /^enlightenment$/) { load_enlightenment_tests(); }
-        if ($de =~ /^mate$/)          { load_mate_tests(); }
-        if ($de =~ /^lxqt$/)          { load_lxqt_tests(); }
-        load_shutdown_tests;
-        return 1;
-    }
-    return 0;
-}
+  load_module($module)
 
-sub load_awesome_tests {
-    loadtest "x11/awesome_menu";
-    loadtest "x11/awesome_xterm";
-}
+Load one test module from a schedule hash. The inclusion/exclusion of the
+module can be controlled by specifying the module with C<module> and the key
+C<only> or <except> next to the module.
+
+We could have this logic to include/exclude specific modules in the modules itself but this would contradict
+https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/150
+even though we still have C<is_applicable> in basetest.pm in os-autoinst.
+
+=cut
 
-sub load_enlightenment_tests {
-    loadtest "x11/enlightenment_first_start";
-    loadtest "x11/terminology";
+sub load_module {
+    my ($module) = @_;
+    return $module unless ref $module eq 'HASH';
+    my %module = %{$module};
+    return $module{module} if ($module{module} && ($module{only} && eval_yaml($module{only})) && !($module{except} && !eval_yaml($module{except})));
```

Comment:
I don't really know how to play golf... even less in perl... yould you try to make this more explicit?. because if one of these conditions fail, it'd be a bit complicated to debug, don't you think?.